### PR TITLE
hub and sites sample notebook

### PR DESCRIPTION
**Pull request by ManushiM**
\<insert pull request description here\>
Sample notebook with a mix of hub and enterprise sites examples.
-----

# Checklist

Please go through each entry in the below checklist and mark an 'X' if that condition has been met. Every entry should be marked with an 'X' to be get the Pull Request approved.

- [ ] All `import`s are in the first cell? 
    - [ ] First block of imports are standard libraries
    - [ ] Second block are 3rd party libraries
    - [ ] Third block are all `arcgis` imports? Note that in some cases, for samples, it is a good idea to keep the imports next to where they are used, particularly for uncommonly used features that we want to highlight.
- [ ] All `GIS` object instantiations are one of the following?
    - `gis = GIS()`
    - `gis = GIS('home')` or `gis = GIS('pro')`
    - `gis = GIS(profile="your_online_portal")`
    - `gis = GIS(profile="your_enterprise_portal")`
- [ ] If this notebook requires setup or teardown, did you add the appropriate code to `./misc/setup.py` and/or `./misc/teardown.py`?
- [ ] If this notebook references any portal items that need to be staged on AGOL/Python API playground, did you coordinate with a Python API team member to stage the item the correct way with the `api_data_owner` user?
- [ ] If the notebook requires working with local data (such as CSV, FGDB, SHP, Raster files), upload the files as items to the [Geosaurus Online Org](geosaurus.maps.arcgis.com/) using `api_data_owner` account and change the notebook to first download and unpack the files.
- [ ] Code simplified & split out across multiple cells, useful comments?
- [ ] Consistent voice/tense/narrative style? Thoroughly checked for typos?
- [ ] All images used like `` instead of ``? All map widgets contain a static image preview? (Call `mapview_inst.take_screenshot()` to do so)
- [ ] All file paths are constructed in an OS-agnostic fashion with `os.path.join()`? (Instead of `r"\foo\bar"`, `os.path.join(os.path.sep, "foo", "bar")`, etc.)
- [ ] Is your code formatted using [Jupyter Black](https://www.freecodecamp.org/news/auto-format-your-python-code-with-black/)? You can use Jupyter Black to format your code in the  notebook.
- [ ] **IF YOU WANT THIS SAMPLE TO BE DISPLAYED ON THE DEVELOPERS.ARCGIS.COM WEBSITE**, ping @ mohi9282 so he can add it to the list for the next deploy

### CODE REVIEW

File: samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb [hunk 1]
```diff
@@ -0,0 +1,787 @@
+{
```
Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <p>In this sample notebook, we will demonstrate how to search for initiatives across a Hub, create a new initiative and edit the layout of its site, link and unlink a page to that site, and clone a site. We will also clone the same site to an Enterprise organization to demonstrate cross-platform cloning of Hub Sites. Finally, we will demonstrate how a site can be fetched using its domain. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <p>In the interest of demonstrating the Hub Premium <code>Initiative</code>, we will sign into a Hub. Sign-in with your ArcGIS credentials while executing this sample notebook to leverage the capabilities of your Hub/Sites licensing.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <p>You can either search for initiatives or filter using parameters like <code>title</code>, <code>owner</code>, or <code>tags</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <p>Linking a page to a site is a two step process. First, you will fetch the page of interest, then you will link it to the site object with a slug for that page.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <p>As we can see, this section has a blue background color. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <p>We can change it to red by assigning the background color value and calling the <code>update_layout</code> method on the site object.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <p>Now the title has successfully been changed. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <p>Sites can be fetched by their <code>itemid</code>. However, in cases where you do not have the <code>siteid</code> for a site that you have access to, you can fetch the item using the following steps:</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <h3>Cloning an initiative/site in the same organization </h3><p> </p><p>We clone an initiative in the same organization by passing the initiative object to the clone method.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <p>Cloning an initiative to Enterprise requires signing into that organization and cloning the site into it. You can clone a site by calling the clone method on the <code>sites</code> property of the target <code>GIS</code> object and providing a reference to the site object and an optional title.  </p><p>Note: Enterprise Sites are supported via the <code>sites</code> submodule of the <code>GIS</code> and not the <code>apps.hub</code> submodule. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <p>Only the site item has been cloned in the new organization as this is an Enterprise organization. The script will automatically determine whether to give you an initiative or a site for your particular Hub.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (BP-Ent) [inline samples/02_power_users_developers/introduction_to_hub_initiatives_and_sites.ipynb:1]: <p>Finally, we will delete the initiative and the site we created.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1451/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

**Review by BP-Ent — CHANGES_REQUESTED** [review]
Suggested changes made on reviewnb

File: guide/13-managing-arcgis-applications/introduction_to_hub_initiatives_and_sites.ipynb [hunk 2]
```diff
@@ -0,0 +1,796 @@
+{
```
Comment (jyaistMap) [inline guide/13-managing-arcgis-applications/introduction_to_hub_initiatives_and_sites.ipynb:1]: Add a cell at the top that:
* is an individual markdown cell with nothing but a single hashmark followed by a title for the notebook

**Review by jyaistMap — CHANGES_REQUESTED** [review]
@ManushiM  - this PR is adding 2 identical notebooks, one as a guide and one as a sample.  Should this be a guide or sample?

**Review by jyaistMap — APPROVED** [review]
Looks great! Thanks @ManushiM 

image

**Pull request merged**